File: chat-service/src/main/java/com/oognuyh/chatservice/repository/UserRepository.java
package com.oognuyh.chatservice.repository;

import com.oognuyh.chatservice.payload.response.UserResponse;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "user-service")
public interface UserRepository {
    
    @GetMapping("/users/{id}")
    UserResponse findUserById(@PathVariable("id") String id);
}

File: notification-service/src/main/kotlin/com/oognuyh/notificationservice/repository/UserRepository.kt
package com.oognuyh.notificationservice.repository

import com.oognuyh.notificationservice.payload.request.StatusUpdateRequest
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.HttpHeaders
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestHeader

@FeignClient(name = "user-service")
interface UserRepository {

    @PutMapping("/v1/users/{id}/status")
    fun updateStatus(
        @RequestHeader(HttpHeaders.AUTHORIZATION) authToken: String,
        @PathVariable("id") id: String,
        statusUpdateRequest: StatusUpdateRequest
    )
}